src/stores/WalletStore.ts

```diff
@@ -61,40 +73,29 @@ export default class WalletStore {
   public login = async (password: string) => {
     this.loading = true;
 
-    // TODO: make this async so it doesnt block the UI
-    // Generate appSalt if needed
-    if (!this.appSalt) {
-      const appSalt: Uint8Array = window.crypto.getRandomValues(new Uint8Array(16)) as Uint8Array;
-      this.appSalt = appSalt;
-      chrome.storage.local.set(
-        { [STORAGE.APP_SALT]: appSalt.toString() },
-        () => console.log('appSalt set'),
-      );
+    // TODO: find solution to unblock UI
+    this.generateAppSalt();
```

Comment:
nitpick. `generate` seems to indicate a new salt each time.

`this.getOrGenerateAppSalt();`

or

`this.getAppSalt()`